runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/DoFnOperator.java

```diff
@@ -258,13 +244,15 @@ public void open() throws Exception {
     pushbackDoFnRunner =
         PushbackSideInputDoFnRunner.create(doFnRunner, sideInputs, sideInputHandler);
 
-    oldDoFn.setup();
+    doFnInvoker = DoFnInvokers.invokerFor(doFn);
+
+    doFnInvoker.invokeSetup();
```

Comment:
It would be somewhat better to call `invokeSetup()` before creating the `DoFnRunner`. Here it should be fine since nothing happens in between, but the convention we would like is that the `DoFn` is ready when it is passed to a `DoFnRunner`.